src/client/java/hadoop-daos/src/main/java/io/daos/fs/hadoop/DaosFileSystem.java

```diff
@@ -238,19 +436,46 @@ public void initialize(URI name, Configuration conf)
       }
 
       // daosFSclient build
-      this.daos = new DaosFsClient.DaosFsClientBuilder().poolId(poolUuid).containerId(contUuid).ranks(svc).build();
-      this.uri = URI.create(name.getScheme() + "://" + name.getAuthority());
-      this.workingDir = new Path("/user", System.getProperty("user.name"))
-              .makeQualified(this.uri, this.getWorkingDirectory());
-      //mkdir workingDir in DAOS
-      daos.mkdir(workingDir.toUri().getPath(), true);
+      DaosFsClient.DaosFsClientBuilder builder = new DaosFsClient.DaosFsClientBuilder().poolId(poolUuid)
+              .containerId(contUuid);
+      if (!StringUtils.isBlank(svrGrp)) {
+        builder.serverGroup(svrGrp);
+      }
+      if (!StringUtils.isBlank(poolFlags)) {
+        builder.poolFlags(Integer.valueOf(poolFlags));
+      }
+      if(!StringUtils.isBlank(svc)) {
+        builder.ranks(svc);
+      }
+      this.daos = builder.build();
+      String tmpUri = name.getScheme() + "://" + name.getAuthority();
+      workPath = "/user/" + System.getProperty("user.name");
+      this.uri = URI.create(tmpUri);
+      if (uns) {
+        qualifiedUnsPrefix = tmpUri + unsPrefix;
+        qualifiedUnsWorkPath = qualifiedUnsPrefix + workPath;
+        workingDir = new Path(qualifiedUnsWorkPath);
+      } else {
+        this.workingDir = new Path(workPath)
+          .makeQualified(this.uri, this.getWorkingDirectory());
+      }
```

Comment:
@zjf2012  i thought this code should be changed as follows, because, the uri of  daos ubs  should not be http://uns:1  instend of  http://uns:1/unsPrefix
`
      if (uns) {
        qualifiedUnsPrefix = tmpUri + unsPrefix;
        qualifiedUnsWorkPath = qualifiedUnsPrefix + workPath;
        this.uri = URI.create(qualifiedUnsPrefix );
        workingDir = new Path(qualifiedUnsWorkPath);
      } else {
        this.uri = URI.create(tmpUri);
        this.workingDir = new Path(workPath)
          .makeQualified(this.uri, this.getWorkingDirectory());
      }
`